Update formatting of step-41.

bangerth opened the pull request:
Specifically:
* Use the same numbering scheme for enumerations as below. The style
  [(1)] likely originated from using an `\item[(1)]` in the original
  latex version of the introduction.
* Add a sentence after the bullet point list at the end of the introduction
  to help doxygen understand that it should not indent the following
  section heading (from the commented program) the same as the bullet
  point list.

drwells commented:
These look a bit strange: on my laptop things are rendered as something like

&middot; [1.]

where I think you meant to use an enumerated list. Would you mind updating this so that we have something like

```
1. Initialize $\mathcal{A}_k and ...
  ...
2. Find the primal-dual pair ...
```

and also apply the same fix to the one below?

drwells commented:
ping @bangerth (this should be pretty easy to fix)

bangerth commented:
The issue was that we start enumerating with zero, not with one. doxygen doesn't know what to do with this. I just changed the itemization so that it starts at one, and updated the only reference I could find to a specific item.

Should be good to go now.

drwells commented:
Since this just changes a `.dox` file we don't really need to run the tester :)
